lib/replicator.js

```diff
@@ -84,6 +87,17 @@ class Attachable {
       this._detach(this.refs[this.refs.length - 1]).reject(err)
     }
   }
+
+  setTimeout (r, ms) {
+    if (r.timeout !== null) {
+      clearTimeout(r.timeout)
+      r.timeout = null
```

Comment:
Maybe `_destroyTimeout(r)` to reuse it above as well?

Reply:
Sure! Can also be static as it doesn’t use this